File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/MarketIntelligence/CatalogosMarketIntelligenceForm.ts
import { LookupEditor, IntegerEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CatalogosMarketIntelligenceForm {
    IdtipoCatalogo: LookupEditor;
    IdClave: IntegerEditor;
    Descripcion: StringEditor;
    Activo: BooleanEditor;
}

export class CatalogosMarketIntelligenceForm extends PrefixedContext {
    static formKey = 'MarketIntelligence.CatalogosMarketIntelligence';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CatalogosMarketIntelligenceForm.init)  {
            CatalogosMarketIntelligenceForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = StringEditor;
            var w3 = BooleanEditor;

            initFormType(CatalogosMarketIntelligenceForm, [
                'IdtipoCatalogo', w0,
                'IdClave', w1,
                'Descripcion', w2,
                'Activo', w3
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/MarketIntelligence/CatalogosMarketIntelligence/CatalogosMarketIntelligenceForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel.DataAnnotations;

namespace MasterDirectory.MarketIntelligence.Forms;

[FormScript("MarketIntelligence.CatalogosMarketIntelligence")]
[BasedOnRow(typeof(CatalogosMarketIntelligenceRow), CheckNames = true)]
public class CatalogosMarketIntelligenceForm
{
    [Editable(false)]
    public int IdtipoCatalogo { get; set; }
    [Editable(false)]
    public int IdClave { get; set; }
    public string Descripcion { get; set; }
    [BooleanEditor]
    public int Activo { get; set; }
    //public DateTime DtRegistro { get; set; }
}